coretex/entities/tag.py

```diff
@@ -0,0 +1,152 @@
+#     Copyright (C) 2023  Coretex LLC
+
+#     This file is part of Coretex.ai
+
+#     This program is free software: you can redistribute it and/or modify
+#     it under the terms of the GNU Affero General Public License as
+#     published by the Free Software Foundation, either version 3 of the
+#     License, or (at your option) any later version.
+
+#     This program is distributed in the hope that it will be useful,
+#     but WITHOUT ANY WARRANTY; without even the implied warranty of
+#     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#     GNU Affero General Public License for more details.
+
+#     You should have received a copy of the GNU Affero General Public License
+#     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from enum import IntEnum
+from typing import Optional, Dict, Any
+from abc import abstractmethod, ABC
+
+import re
+import random
+import logging
+
+from ..networking import networkManager, NetworkRequestError
+
+
+class EntityTagType(IntEnum):
+
+    model   = 1
+    dataset = 2
+
+
+class Taggable(ABC):
+
+    id: int
+    projectId: int
+
+    @property
+    @abstractmethod
+    def entityTagType(self) -> int:
+        pass
+
+    def _getTagId(self, tagName: str) -> Optional[int]:
+        parameters = {
+            "name": tagName,
+            "type": int(self.entityTagType),
```

Comment:
isn't entityTagType already of type int? why is casting to int necessary here?

Reply:
Backend receives a EntityTagType class for the get request (which doesn't happen for post requests) so I cast it to int even though it behaves like an integer object in python code.

Reply:
Why not just change the return type of `entityTagType` to `EntityTagType` and then instead of casting it to `int` here you can do it like `self.entityTagType.value`.

Reply:
The exact value return value of `EntityTagType.dataset` is `Literal[EntityTagType.dataset]` and similarly `.model` for model. How about we instead keep `int` as the return type of `entityTagType` and return EntityTagType.dataset.value / .model.value.
This way we keep a standardized return value for `entityTagType`. All uses of this property need the enum integer value anyway.

Reply:
But that would allow returning any integer, it's better to use enums to limit the number of available values which can be returned to prevent potential errors during coding. If you put `int` as the return type you can return any kind of integer for that, whilst if the return type of the property is `EntityTagType` mypy will complain if you do not return a valid value from that enum.